dq/src/main/java/com/wipro/ats/bdre/dq/DQMain.java

```diff
@@ -35,24 +35,27 @@
      * @throws Exception
      */
     public static void main(String[] args) throws Exception {
-        CommandLine commandLine = new DQMain().getCommandLine(args, PARAMS_STRUCTURE);
-        String processId = commandLine.getOptionValue("process-id");
-        String sPath = commandLine.getOptionValue("source-file-path");
-        String destDir = commandLine.getOptionValue("destination-directory");
-
-        int result=0;
-        if(sPath.indexOf(';')!=-1 || sPath.indexOf(',')!=-1){
-            String[] lof = sPath.split(";");
-            String entries[] = lof[0].split(",");
-            String[] params = {processId, entries[2], destDir};
-            result = ToolRunner.run(new Configuration(),new DQDriver(), params);
-
-        }else{
-            String[] params = {processId, sPath, destDir};
-            result = ToolRunner.run(new Configuration(),new DQDriver(), params);
+        try {
+            CommandLine commandLine = new DQMain().getCommandLine(args, PARAMS_STRUCTURE);
+            String processId = commandLine.getOptionValue("process-id");
+            String sPath = commandLine.getOptionValue("source-file-path");
+            String destDir = commandLine.getOptionValue("destination-directory");
+
+            int result = 0;
+            if (sPath.indexOf(';') != -1 || sPath.indexOf(',') != -1) {
+                String[] lof = sPath.split(";");
+                String[] entries = lof[0].split(",");
+                String[] params = {processId, entries[2], destDir};
+                result = ToolRunner.run(new Configuration(), new DQDriver(), params);
+
+            } else {
+                String[] params = {processId, sPath, destDir};
+                result = ToolRunner.run(new Configuration(), new DQDriver(), params);
+            }
+        }
+        finally {
+            throw new Exception("dq exit exception");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not throw exceptions in finally blocks. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1163)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00112)